File: Programación Orientada a Objetos (PDOO)/Napakalaki(Java)/src/napakalaki/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package napakalaki;
import java.util.*;
import java.util.Collections;

/**
 *
 * @author Josele
 */
public class CultistPlayer extends Player{
    private static int totalCultistPlayers=0;
    private Cultist myCultistCard;
    
    public CultistPlayer(Player p, Cultist c){
        super(p);
        //enemigo=p.getEnemy();      
        myCultistCard=c;
        totalCultistPlayers++;
    }
     
    @Override
    public int getCombatLevel(){
        float nivel=super.getCombatLevel();
        float lvl= (nivel+ (1/5* nivel))+ myCultistCard.getGainedLevels()*totalCultistPlayers;
        lvl=lvl*100;
        lvl=java.lang.Math.round(lvl);
        lvl=lvl/100;
        
        return (int)lvl;
    }
    @Override
    protected int getOponentLevel(Monster m){
        return m.getCombatLevelAgainstCultistPlayer();
    }
    @Override
    protected boolean shouldConvert(){
        return false;
    }
    @Override
    protected Treasure giveMeATreasure(){
          
        Random rnd = new Random();
        return this.getVisibleTreasures().get(rnd.nextInt(this.getVisibleTreasures().size()));
        
    }
    
    @Override
    protected boolean canYouGiveMeATreasure(){
        int tama= this.getVisibleTreasures().size();
        if(tama>0){
            return true;
        }
        else{
            return false;
        }
    }
  
    public int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }
}

File: Programación Orientada a Objetos (PDOO)/Napakalaki(Ruby)/pkg/pruebaNapakalaki-0.0.1/lib/cultist_player.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "player.rb"
module NapakalakiGame
class CultistPlayer < Player
  attr_accessor :my_cultist_card
  @@total_cultits_players=0
  
  
  def initialize(p,c)
    #copiarJugador(p)
    super(p.name)
    @my_cultist_card=c
    @@total_cultits_players+=1
  end
  
  def getCombatLevel
    nivel= super
    lvl= (nivel+(1/5*nivel))+ @my_cultist_card.ganied_levels*@@total_cultist_players
    lvl = lvl.round
    lvl
  end
  
  def getOponentLevel(m)
    return m.getCombatLevelAgainstCultistPlayer
  end
  
  def shouldConvert
    return false
  end
  
  def giveMeATreasure
    rnd = rand(@visibleTreasures.size)
    return @visibleTreasures.get(rand)
  end
  
  def canYouGiveMeATreasure
    tama = @visibleTreasures.size
    if tama > 0 then
      return true
    else
      return false
    end
  end
  
  def self.getTotalCultistPlayers
    return @@total_cultist_players
  end
end
end